test/extensions/filters/network/common/redis/client_impl_test.cc

```diff
@@ -810,6 +810,60 @@ TEST_F(RedisClientImplTest, MovedRedirectionNotEnabled) {
   client_->close();
 }
 
+TEST_F(RedisClientImplTest, RemoveFailedHealthCheck) {
```

Comment:
nit: can you add a small `//` comment on top of each new test that briefly describes what the test does since it's non-obvious?